### DISCUSSION TITLE
Use "Pick" for type in auto generated mutations?

### DISCUSSION THREAD

**Discussion by cktang88** [blitz-js/blitz#782] [General]
Currently in `app/projects/mutations/updateProject.ts` for example, the type is something like:
```ts
type UpdateProjectInput = {
  where: ProjectUpdateArgs["where"]
  data: ProjectUpdateArgs["data"]
}
```

Instead I think it'd be cleaner just to do:
```ts
type UpdateProjectInput = Pick<ProjectUpdateArgs, "where" | "data">
```

**Comment by flybayer** [comment]
Hey @cktang88 thanks for opening this!

I can definitely see how that seems a bit nicer.

One downside, is that you have to then refactor if you want to do something like this:

```ts
type UpdateProjectInput = {
  where: ProjectUpdateArgs["where"]
  data: Omit<ProjectUpdateArgs["data"], "id">
}
```

I'm curious to know what others think!

---

By the way, once [zod](https://github.com/vriad/zod) adds support or matching existing types, I think it will be a lot better (and more secure!) to scaffold queries and mutations by default with input validation like this:

```ts
import db, {FindOneUserArgs} from "db"
import {SessionContext} from "blitz"
import * as z from "zod"

export const GetUserInput: z.type<Pick<FindOneUserArgs, "where">> = z.object({
  where: z.object({
    id: z.string()
  })
})
export type GetUserInputType = z.infer<typeof GetUserInput>


export default async function getUser(
  input: GetUserInput,
  ctx: {session?: SessionContext} = {},
) {
  const {id} = GetUserInput.parse(input)
  // stuff
}
```

If people want, we could also add `yup` as an option.
